vividus-plugin-rest-api/src/main/java/org/vividus/http/validation/ResourceValidator.java

```diff
@@ -71,8 +73,9 @@ private void validateResource(URI uri, T resourceValidation)
     {
         try
         {
-            int statusCode = httpClient.doHttpHead(uri).getStatusCode();
-            if (notAllowedHeadStatusCodes.contains(statusCode))
+            HttpResponse response = showBody ? httpClient.doHttpGet(uri) : httpClient.doHttpHead(uri);
```

Comment:
I don't think we need to change the logic here, we should always try to run HTTP HEAD. Majority of cases with failures requiring investigation use HTTP GET fall-back.